connectdialog.cpp

```diff
@@ -26,20 +26,26 @@ ConnectDialog::ConnectDialog(QWidget *parent) :
     // fill the combo box values
     fillSettingsLists();
 
-    // define the default values
-
     // for device take 1st device listed (if any)
     QString default_device;
     if (ui->deviceList->count() > 0)
         default_device = ui->deviceList->itemText(0);
-    defaultValues["device"] = default_device;
-    defaultValues["baud_rate"] = QString::number(QSerialPort::Baud115200);
-    defaultValues["data_bits"] = QString::number(QSerialPort::Data8);
-    defaultValues["stop_bits"] = QString("1");
-    defaultValues["parity"] = QString("None");
-    defaultValues["flow_control"] = QString("None");
-
-    preselectPortSettings(defaultValues);
+
+    // define a default configuration
+    QHash<QString, QString> default_cfg;
+    default_cfg["device"] = default_device;
+    default_cfg["baud_rate"] = QString::number(QSerialPort::Baud115200);
+    default_cfg["data_bits"] = QString::number(QSerialPort::Data8);
+    default_cfg["stop_bits"] = QString("1");
```

Comment:
E' una piccolezza e me ne sono ricordato ora, ma in questi casi si può usare:
http://doc.qt.io/qt-5/qstring.html#QStringLiteral